### ISSUE TITLE
`EntityAssigner.assign` fails to serialize `ObjectId`s correctly when using `mergeObjects: true`

### ISSUE THREAD

**Issue by alex-scown-checkit** [mikro-orm/mikro-orm#365]
**Describe the bug**
`EntityAssigner.assign` fails to serialize `ObjectId`s correctly when using `mergeObjects: true`. The root cause is that `Buffer`s are merged key by key and cease to be `Buffer`s in the process.

**Stack trace**
```
 Error: object [{"_bsontype":"ObjectID","id":{"0":94,"1":86,"2":87,"3":203,"4":114,"5":125,"6":17,"7":200,"8":103,"9":211,"10":14,"11":94}}] is not a valid ObjectId
      at serializeObjectId (node_modules/bson/lib/bson/parser/serializer.js:287:11)
      at serializeInto (node_modules/bson/lib/bson/parser/serializer.js:931:17)
      at serializeObject (node_modules/bson/lib/bson/parser/serializer.js:347:18)
      at serializeInto (node_modules/bson/lib/bson/parser/serializer.js:937:17)
      at serializeObject (node_modules/bson/lib/bson/parser/serializer.js:347:18)
      at serializeInto (node_modules/bson/lib/bson/parser/serializer.js:937:17)
      at serializeObject (node_modules/bson/lib/bson/parser/serializer.js:347:18)
      at serializeInto (node_modules/bson/lib/bson/parser/serializer.js:727:17)
      at serializeObject (node_modules/bson/lib/bson/parser/serializer.js:347:18)
      at serializeInto (node_modules/bson/lib/bson/parser/serializer.js:937:17)
      at BSON.serialize (node_modules/bson/lib/bson/bson.js:64:28)
      at Msg.serializeBson (node_modules/mongodb/lib/core/connection/msg.js:125:22)
      at Msg.makeDocumentSegment (node_modules/mongodb/lib/core/connection/msg.js:117:33)
      at Msg.toBin (node_modules/mongodb/lib/core/connection/msg.js:103:25)
      at MessageStream.writeCommand (node_modules/mongodb/lib/cmap/message_stream.js:80:28)
      at Connection.write (node_modules/mongodb/lib/cmap/connection.js:353:26)
      at _command (node_modules/mongodb/lib/core/wireprotocol/command.js:128:10)
      at command (node_modules/mongodb/lib/core/wireprotocol/command.js:28:5)
      at writeCommand (node_modules/mongodb/lib/core/wireprotocol/write_command.js:47:3)
      at Object.insert (node_modules/mongodb/lib/core/wireprotocol/index.js:6:5)
      at Connection.insert (node_modules/mongodb/lib/cmap/connection.js:178:8)
      at server.s.pool.withConnection (node_modules/mongodb/lib/core/sdam/server.js:486:13)
      at Object.checkOut (node_modules/mongodb/lib/cmap/connection_pool.js:347:7)
      at createConnection (node_modules/mongodb/lib/cmap/connection_pool.js:497:23)
      at connect (node_modules/mongodb/lib/cmap/connection_pool.js:427:7)
      at callback (node_modules/mongodb/lib/core/connection/connect.js:97:5)
      at runCommand (node_modules/mongodb/lib/core/connection/connect.js:169:5)
      at conn.command (node_modules/mongodb/lib/core/connection/connect.js:334:7)
      at MessageStream.messageHandler (node_modules/mongodb/lib/cmap/connection.js:262:5)
      at MessageStream.EventEmitter.emit (domain.js:448:20)
      at processMessage (node_modules/mongodb/lib/cmap/message_stream.js:140:12)
      at MessageStream._write (node_modules/mongodb/lib/cmap/message_stream.js:66:7)
      at doWrite (_stream_writable.js:415:12)
      at writeOrBuffer (_stream_writable.js:399:5)
      at MessageStream.Writable.write (_stream_writable.js:299:11)
      at Socket.ondata (_stream_readable.js:708:20)
      at Socket.EventEmitter.emit (domain.js:448:20)
      at addChunk (_stream_readable.js:287:12)
      at readableAddChunk (_stream_readable.js:268:11)
      at Socket.Readable.push (_stream_readable.js:223:10)
      at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

**To Reproduce**
1. Create the following entity:
```typescript
interface NestedProperty {
  nested: {
    id: ObjectId;
  };
}

@Entity({collection: 'test-entities'})
class TestEntity implements AnyEntity<TestEntity, '_id'> {
  @PrimaryKey()
  public _id!: string;

  @Property()
  public p: NestedProperty = {} as NestedProperty;

}
```
2. Persist the entity as follows:
```typescript
    const orm = await MikroORM.init({
      entities: [TestEntity],
      clientUrl: 'mongodb://localhost:27017',
      dbName: 'test-entities',
      metadataProvider: ReflectMetadataProvider
    });

    const em = orm.em;

    const entity = new TestEntity();

    wrap(entity).assign({
      _id: uuid(),
      p: {
        nested: {
          id: new ObjectId()
        }
      }
    }, {
      mergeObjects: true,
      em
    });

    await em.persistAndFlush(entity);
```

**Expected behavior**
The entity should be persisted to MongoDB.

**Versions**

| Dependency | Version |
| - | - |
| node | 10.17.0 |
| typescript | 3.7.5 |
| mikro-orm | 3.1.0 |
| mongodb | 3.5.2 |

**Comment by B4nan** [comment]
available in v3.3.2

**Comment by alex-scown-checkit** [comment]
Thanks!
